src/cuOpt.jl

```diff
@@ -36,6 +36,35 @@ function __init__()
         )
     end
     global libcuopt = Libdl.dlpath(libname)
+
+    # Check library version
+    version_major = Ref{Int32}(0)
+    version_minor = Ref{Int32}(0)
+    version_patch = Ref{Int32}(0)
+
+    # Note: We need to call cuOptGetVersion after libcuopt is set
+    status = ccall(
+        (:cuOptGetVersion, libcuopt),
+        Int32,
+        (Ptr{Int32}, Ptr{Int32}, Ptr{Int32}),
+        version_major,
+        version_minor,
+        version_patch,
+    )
+
+    if status != 0
+        error("Failed to get cuOpt library version (status code: $status)")
+    end
+
+    expected_major = 25
+    expected_minor = 10
+
+    if version_major[] != expected_major || version_minor[] != expected_minor
```

Comment:
IT's a bit hard to tell from the diff of `libcuopt.jl`, but was there a breaking change to the C API? Can the old version still be used?

Reply:
Yes, old version should still support it, but just for better performance and latest, thought of switching to latest version. Please let me know if I need to change it so that <= 25.10 ?